libs/kinematics/src/CVehicleSimulVirtualBase.cpp

```diff
@@ -24,24 +26,59 @@ CVehicleSimulVirtualBase::~CVehicleSimulVirtualBase()
 
 void CVehicleSimulVirtualBase::setCurrentGTPose(const mrpt::math::TPose2D  &pose)
 {
-	m_pose=pose;
+	m_GT_pose=pose;
 }
 
 void CVehicleSimulVirtualBase::simulateOneTimeStep(const double dt)
 {
+	using mrpt::poses::CPose2D;
 	const double final_t = m_time + dt;
 	while (m_time <= final_t)
 	{
-		this->internal_simulStep(m_firmware_control_period);
+        // Simulate movement during At:
+		CPose2D nextOdometry = CPose2D(m_odometry) +
+			CPose2D (
+				m_odometric_vel.vx*m_firmware_control_period,
+				m_odometric_vel.vy*m_firmware_control_period,
+				m_odometric_vel.omega*m_firmware_control_period
+			);
+		CPose2D nextGT = CPose2D(m_GT_pose) +
+			CPose2D(
+				m_GT_vel.vx*m_firmware_control_period,
+				m_GT_vel.vy*m_firmware_control_period,
+				m_GT_vel.omega*m_firmware_control_period
+			);
+		mrpt::math::wrapToPi(nextOdometry.phi());
```

Comment:
wrapToPi() returns a value, does not modify a ref as wrapTo2PiInPlace()...

please ==> p.phi( wrapTo2Pi(p.phi()) );